sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,163 @@
+package azcosmos
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+)
+
+const globalEndpointManagerDefaultBackgroundRefreshInterval = 5 * time.Minute
+
+type globalEndpointManager struct {
+	client           *Client
+	preferredRegions []string
+	locationCache    *locationCache
+	refreshTicker    *time.Ticker
+	refreshDone      chan struct{}
+	sync.Mutex
+}
+
+// NewGlobalEndpointManager creates a new global endpoint manager.
+func newGlobalEndpointManager(client *Client, preferredRegions []string, refreshInterval time.Duration) (*globalEndpointManager, error) {
```

Comment:
From where will this be called?  I think we need to know the answer to that in order to help control the lifetime of the refresh goroutine.

Reply:
@jay-most please cover expected interactions with GlobalEnpointManager part of PR description. Few ones of interest
- How is GlobalEnpointManager created and lifetime managed
- How is GlobalEnpointManager leveraged in request-flows 
> - First region choice of routing 
> - How are regions status managed/maintained
> - On-demand account refresh scenarios

Reply:
I was discussing this with Jeff and clients must _not_ have a close method.  This also goes against our client [design guidelines](https://azure.github.io/azure-sdk/golang_introduction.html#client-immutability) that state clients must not create any goroutines (this point isn't very clear, we will get this fixed).